src/HttpClientRequestAdapter.cs

```diff
@@ -324,7 +324,7 @@ private async Task DrainAsync(HttpResponseMessage response)
         }
         private bool shouldReturnNull(HttpResponseMessage response)
         {
-            return response.StatusCode == HttpStatusCode.NoContent;
+            return response.StatusCode == HttpStatusCode.NoContent || response.Content == null;

```

Comment:
calling the attention to this IDisposable implementation static analysis comment (details in sonarqube)

Reply:
Refactored the Dispose function as suggested